ui/src/app/navigation/commoncontrols.component.ts

```diff
@@ -14,10 +17,30 @@ export class CommoncontrolsComponent extends AbstractSearchComponent implements
   searchIcon = SvgHelper.path(SvgIcon.SEARCH)
   dismissIcon = SvgHelper.path(SvgIcon.DISMISS)
 
-  constructor(protected searchService: SearchService, protected route: ActivatedRoute, protected authService: AuthService) {
+  constructor(
+    protected searchService: SearchService,
+    protected route: ActivatedRoute,
+    protected authService: AuthService,
+    protected richSkillService: RichSkillService,
+    @Inject(LOCALE_ID) protected locale: string) {
```

Comment:
You're right.  Adding so many dependencies implies to me that the file wasn't intended to address the new code.  Perhaps the export code belongs somewhere else?  Could it be put in a LibraryExportComponent.ts?

Reply:
Or if it just supports the rich-skill.service.ts, then perhaps the new code belongs there.  Just thinking about it is all.